src/Microsoft.OData.Edm/Schema/EdmModelBase.cs

```diff
@@ -23,6 +24,11 @@ public abstract class EdmModelBase : EdmElement, IEdmModel
         private readonly Dictionary<string, IEdmTerm> termDictionary = new Dictionary<string, IEdmTerm>();
         private readonly Dictionary<string, IList<IEdmOperation>> functionDictionary = new Dictionary<string, IList<IEdmOperation>>();
 
+        /// <summary>
+        /// Cache of operations that are bindable to entity types.
+        /// </summary>
+        private readonly ConcurrentDictionary<string, IList<IEdmOperation>> bindableOperationsCache;

```

Comment:
>bindableOperationsCache [](start = 76, length = 23)

What's the difference between "functionDictionary" and new "bindableOperationsCache"?

Reply:
functionDictionary is all functions, indexed by name. This is a cache of all bindable functions, indexed by binding type.  could perhaps make name or comment clearer?

---
In reply to: [492997426](https://github.com/OData/odata.net/pull/1888#discussion_r492997426) [](ancestors = 492997426)

Reply:
key: full type name of the binding type. 

Key in the functionDictionary is the full operation name.

---
In reply to: [492997426](https://github.com/OData/odata.net/pull/1888#discussion_r492997426) [](ancestors = 492997426)

Reply:
If it's binding to collection, the key is string like "Collection(Namespace.Customer)" ?

---
In reply to: [493025946](https://github.com/OData/odata.net/pull/1888#discussion_r493025946) [](ancestors = 493025946,492997426)

Reply:
Had tested using  a test , fulltypename coming as
bindingType.FullTypeName()
"Collection(TestModel.Movie)"